### DISCUSSION TITLE
Longhorn volume size

### DISCUSSION THREAD

**Discussion by transmogrifix** [mysticaltech/terraform-hcloud-kube-hetzner#682] [Q&A] [answered]
Hi,

I have the following configuration in my kube.tf file:
```
agent_nodepools = [
    { # Node for TimescaleDB with 500GB volume.
      name        = "agent-cx41-l500",
      server_type = "cx41",
      location    = "nbg1",
      labels      = [
        "node.kubernetes.io/server-usage=tsdb",
        "node.kubernetes.io/storage-type=longhorn",
        "node.kubernetes.io/longhorn_volume_size=500"
      ],
      taints      = [],
      count       = 1,
      longhorn_volume_size = 500
    },
    { # A general node for application pods and PostgresSQL with local NVME storage.
      name        = "agent-cx51-l0",
      server_type = "cx51",
      location    = "nbg1",
      labels      = [
        "node.kubernetes.io/server-usage=general",
        "node.kubernetes.io/storage-type=longhorn",
        "node.kubernetes.io/longhorn_volume_size=0",
        "node.longhorn.io/create-default-disk=config"        
      ],
      taints      = [],
      count       = 1,
      longhorn_volume_size = 0
    }
  ]
  enable_longhorn = true
```
I was under the impression that when I deploy, node from the first pool will get a 500GB Hetzner volume created and attached. What actually happens is longhorn uses local node disks and no Hetzner volume gets created.

I have tried with versions 1.10.9 (some others from 1.9.x range as well but not sure which ones exactly) and 2.0.0.

I have searched through the issues and discussions on this project and tried to follow the excellent best practices gist https://gist.github.com/ifeulner/d311b2868f6c00e649f33a72166c2e5b by ifeulner but I can't seem to get a volume created and used automatically.

I have worked around the problem by creating a volume manually, attached it to the node and added it manually via the Longhorn UI. It works fine this way but I would really like to avoid this.

Any ideas or suggestions?

Thank you!

**Comment by mysticaltech** [comment]
@transmogrifix This is the code taking care of this:

https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/192866e0dea1a1135ce013614254550694c75306/agents.tf#L99-L142

 It should work and was working in the past. I will test myself and let you know.

**Comment by mysticaltech** [comment, marked as answer]
@transmogrifix Thanks for finding this bug! It's been fixed in v2.0.1. You should be able to just `terraform init -upgrade` to install the missing volumes and config.

**Reply by transmogrifix** [reply]
Thank you for fixing it so fast!

Also a big thank you to everyone who contributed to this great project!

**Comment by valkenburg-prevue-ch** [comment]
Oh wow, and here I was wondering why ssd storage seemed just as fast as node storage (and why longhorn storage size did not reflect the ssd size)....! 
I guess this was the fix? https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/commit/47dc93cdcde242200529563b7e56a2161f052cc4

**Reply by mysticaltech** [reply]
@valkenburg-prevue-ch Exactly!
